src/core/control/tools/VerticalToolHandler.cpp

```diff
@@ -155,27 +161,46 @@ bool VerticalToolHandler::onKeyReleaseEvent(GdkEventKey* event) {
 
 auto VerticalToolHandler::getElements() const -> const std::vector<Element*>& { return this->elements; }
 
+auto VerticalToolHandler::getElementsBoundingRect() const -> std::optional<xoj::util::Rectangle<double>> {
+    if (this->elements.empty()) {
+        return std::nullopt;
+    }
+    xoj::util::Rectangle<double> rect;
+    for_first_then_each(
+            this->elements, [&](Element* e) { rect = e->boundingRect(); },
+            [&](Element* e) { rect.unite(e->boundingRect()); });
+    return rect;
+}
+
 auto VerticalToolHandler::finalize() -> std::unique_ptr<MoveUndoAction> {
+    if (this->elements.empty()) {
+        auto [min, max] = std::minmax(this->startY, this->endY);
+        this->view->repaintRect(0, min, this->page->getWidth(), max - min);
+        return nullptr;
+    }
     double dY = this->endY - this->startY;
 
     auto undo =
             std::make_unique<MoveUndoAction>(this->layer, this->page, &this->elements, 0, dY, this->layer, this->page);
 
+    std::optional<xoj::util::Rectangle<double>> rect = this->getElementsBoundingRect();
+    assert(rect);
```

Comment:
nit: `g_assert` gives a better error message.

Also, if you move up the rect line right after the `this->elements->empty()` block, I think it will be clearer that rect should not be the nullopt.